src/view/form/editing-form-view.js

```diff
@@ -1,43 +1,111 @@
   #point = null;
   #pointDestination = null;
-  #pointOffers = null;
+  #arrayOffers = null;
   #arrayDestinationsModel = null;
   #rollupClickHandler = null;
   #onSubmitClick = null;
 
-  constructor({point, pointDestination, pointOffers, arrayDestinationsModel, onRollupClick, onSubmitClick}){
+  constructor({point, pointDestination, arrayOffers, arrayDestinationsModel, onRollupClick, onSubmitClick}){
     super();
-    this.#point = point;
+    this._setState(EditingFormView.parsePointToState({point}));
     this.#pointDestination = pointDestination;
-    this.#pointOffers = pointOffers;
+    this.#arrayOffers = arrayOffers;
     this.#arrayDestinationsModel = arrayDestinationsModel;
     this.#rollupClickHandler = onRollupClick;
     this.#onSubmitClick = onSubmitClick;
-    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#rollupButtonClickHandler);
-    this.element.querySelector('.event__save-btn').addEventListener('click', this.#pointSubmitHandler);
+
+    this._restoreHandlers();
   }
 
   get template(){
     return createFormTemplate({
-      point: this.#point,
+      state: this._state,
       pointDestination: this.#pointDestination,
-      pointOffers: this.#pointOffers,
+      arrayOffers: this.#arrayOffers,
       arrayDestinationsModel: this.#arrayDestinationsModel,
     });
   }
 
+  reset(point) {
+    this.updateElement({point});
+  }
+
+  _restoreHandlers = () => {
+    this.element.querySelector('.event__input--destination').addEventListener('change',this.#destinationChangeHandler);//
+    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#rollupButtonClickHandler);//
+    this.element.querySelector('.event__save-btn').addEventListener('click', this.#pointSubmitHandler);
+    this.element.querySelector('.event__input--price').addEventListener('change',this.#priceChangeHandler);//
+    this.element.querySelector('.event__type-group').addEventListener('change',this.#typeChangeHandler);//
+    this.element.querySelector('.event__available-offers').addEventListener('change',this.#offersChangeHandler);//
+  };
+
+  #typeChangeHandler = (evt) => {
+    evt.preventDefault();
+    this.updateElement({
+      point: {
+        ...this._state.point,
+        type:evt.target.value,
+        offers: [],
+      }
+    });
+  };
+
+  #offersChangeHandler = () => {
+    const checkedBoxes = Array.from(this.element.querySelectorAll('.event__offer-checkbox:checked'));
+
+    this._setState({
+      point: {
+        ...this._state.point,
+        offers:checkedBoxes.map((element) => element.dataset.offerId)
+      }
+    });
+  };
+
+  #destinationChangeHandler = (evt) => {
+    evt.preventDefault();
+    const selectedDistination = this.#arrayDestinationsModel.find((elem) => elem.name === evt.target.value);
+    const selectedDistinationId = (selectedDistination) ? selectedDistination.id : null;
+    if (selectedDistination === undefined) {
+      return;
```

Comment:
здесь луше верещ отрицание делать
```suggestion
    if (!selectedDistination) {
      return;
```